diff --git a/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php b/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php
index 6206b23fb..6e81cc64f 100644
--- a/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php
+++ b/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php
@@ -373,6 +373,21 @@ abstract class AbstractDb extends AbstractResource
      */
     public function save(\Magento\Framework\Model\AbstractModel $object)
     {
+        if ($object instanceof \Magento\Customer\Model\Address\AbstractAddress) {
+            if ($object->getAddressType() == 'shipping' && $object->getShippingMethod() && $object->getQuote()->getStore()->getCode() == 'my_laneige'
+                && (!$object->getFirstname() || !$object->getLastname()))
+            {
+                $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+                $backTrace = '';
+                foreach ($debugBackTrace as $debug) {
+                    $backTrace .= @$debug['class'] . ':' . @$debug['line'] . @$debug['type'] . @$debug['function'] . " | ";
+                }
+                $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/log-error-BRANDSR-5507.log');
+                $logger = new \Zend_Log();
+                $logger->addWriter($writer);
+                $logger->info('Missing shipping before save address Track Trace: ' . $backTrace);
+            }
+        }
         if ($object->isDeleted()) {
             return $this->delete($object);
         }
diff --git a/vendor/magento/module-quote/Model/QuoteManagement.php b/vendor/magento/module-quote/Model/QuoteManagement.php
index 1933b5614..74541f8be 100644
--- a/vendor/magento/module-quote/Model/QuoteManagement.php
+++ b/vendor/magento/module-quote/Model/QuoteManagement.php
@@ -437,8 +437,19 @@ class QuoteManagement implements CartManagementInterface
         }

         $this->eventManager->dispatch('checkout_submit_before', ['quote' => $quote]);
+        $shippingAddress = $quote->getShippingAddress();
+        if ($quote->getStore()->getCode() == 'my_laneige' && !$shippingAddress->getFirstname() || !$shippingAddress->getLastname()) {
+            $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            $backTrace = '';
+            foreach ($debugBackTrace as $debug) {
+                $backTrace .= @$debug['class'] . ':' . @$debug['line'] . @$debug['type'] . @$debug['function'] . " | ";
+            }
+            $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/log-error-BRANDSR-5507.log');
+            $logger = new \Zend_Log();
+            $logger->addWriter($writer);
+            $logger->info('Missing shipping before submit quote Track Trace: ' . $backTrace);
+        }
         $order = $this->submit($quote);
-
         if (null == $order) {
             throw new LocalizedException(
                 __('A server error stopped your order from being placed. Please try to place your order again.')
