diff --git a/vendor/magento/module-quote/Model/Quote.php b/vendor/magento/module-quote/Model/Quote.php
index f5224abbc..ccf1fcbd4 100644
--- a/vendor/magento/module-quote/Model/Quote.php
+++ b/vendor/magento/module-quote/Model/Quote.php
@@ -696,6 +696,18 @@ class Quote extends AbstractExtensibleModel implements \Magento\Quote\Api\Data\C
      */
     public function setReservedOrderId($reservedOrderId)
     {
+        if (($this->getReservedOrderId() && $this->getReservedOrderId() != $reservedOrderId) || !$reservedOrderId) {
+            $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/linepay.log');
+            $logger = new \Zend_Log();
+            $logger->addWriter($writer);
+            $logger->crit('Re-set Reserved Order Id for Quote:' . $this->getIncrementId() . ' with payment: ' . $this->getPayment()->getMethod());
+            $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            $backTrace = '';
+            foreach ($debugBackTrace as $debug) {
+                $backTrace .= @$debug['class'] . ':' . @$debug['line'] . @$debug['type'] . @$debug['function'];
+            }
+           $logger->crit('Back trace: ' . $backTrace);
+        }
         return $this->setData(self::KEY_RESERVED_ORDER_ID, $reservedOrderId);
     }

diff --git a/vendor/magento/module-quote/Model/QuoteManagement.php b/vendor/magento/module-quote/Model/QuoteManagement.php
index 60cd1c9b7..d3566d8e0 100644
--- a/vendor/magento/module-quote/Model/QuoteManagement.php
+++ b/vendor/magento/module-quote/Model/QuoteManagement.php
@@ -7,6 +7,7 @@ declare(strict_types=1);

 namespace Magento\Quote\Model;

+use Eguana\LinePay\Model\LinePayLogger;
 use Magento\Authorization\Model\UserContextInterface;
 use Magento\Checkout\Model\Session as CheckoutSession;
 use Magento\Customer\Api\AccountManagementInterface;
@@ -224,6 +225,7 @@ class QuoteManagement implements CartManagementInterface
         CustomerSession $customerSession,
         AccountManagementInterface $accountManagement,
         QuoteFactory $quoteFactory,
+        LinePayLogger $linePayLogger,
         QuoteIdMaskFactory $quoteIdMaskFactory = null,
         AddressRepositoryInterface $addressRepository = null,
         RequestInterface $request = null,
@@ -249,6 +251,7 @@ class QuoteManagement implements CartManagementInterface
         $this->accountManagement = $accountManagement;
         $this->customerSession = $customerSession;
         $this->quoteFactory = $quoteFactory;
+        $this->linePayLogger = $linePayLogger;
         $this->quoteIdMaskFactory = $quoteIdMaskFactory ?: ObjectManager::getInstance()
             ->get(QuoteIdMaskFactory::class);
         $this->addressRepository = $addressRepository ?: ObjectManager::getInstance()
@@ -535,7 +538,16 @@ class QuoteManagement implements CartManagementInterface
             $this->customerManagement->populateCustomerInfo($quote);
         }
         $addresses = [];
+        if ($quote->getPayment()->getMethod() == 'linepay_payment') {
+            $linepayReservedOrderId = $quote->getReservedOrderId();
+            $orderSendToLinePay = $this->checkoutSession->getData('linepay_order_id');
+            if (!$linepayReservedOrderId || $linepayReservedOrderId != $orderSendToLinePay) {
+                $this->linePayLogger->addAPICallLog('order_id is diffirent, current order_id= ' . $linepayReservedOrderId . ' linepay_order_id= ' . $orderSendToLinePay,  []);
+            }
+
+        }
         $quote->reserveOrderId();
+        if ($quote->getPayment())
         if ($quote->isVirtual()) {
             $this->dataObjectHelper->mergeDataObjects(
                 OrderInterface::class,
