diff --git a/vendor/amasty/module-page-speed-tools/Lib/MobileDetect.php b/vendor/amasty/module-page-speed-tools/Lib/MobileDetect.php
index 5f138659c..732359366 100644
--- a/vendor/amasty/module-page-speed-tools/Lib/MobileDetect.php
+++ b/vendor/amasty/module-page-speed-tools/Lib/MobileDetect.php
@@ -1473,7 +1473,12 @@ class MobileDetect
             return false;
         }

-        $match = (bool) preg_match(sprintf('#%s#is', $regex), (false === empty($userAgent) ? $userAgent : $this->userAgent), $matches);
+        $agent = $this->userAgent;
+        if (false === empty($userAgent)) {
+        $agent = $userAgent;
+        }
+
+        $match = (bool) preg_match(sprintf('#%s#is', $regex), ($agent ? $agent : ''), $matches);
         // If positive match is found, store the results for debug.
         if ($match) {
             $this->matchingRegex = $regex;
