Subject: [PATCH] Create path fix preg_match
---
Index: vendor/amasty/rewards/Block/Adminhtml/Sales/Creditmemo.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/amasty/rewards/Block/Adminhtml/Sales/Creditmemo.php b/vendor/amasty/rewards/Block/Adminhtml/Sales/Creditmemo.php
--- a/vendor/amasty/rewards/Block/Adminhtml/Sales/Creditmemo.php	(revision 52e6cc80378f3e1121ce1d21e755ece04871568b)
+++ b/vendor/amasty/rewards/Block/Adminhtml/Sales/Creditmemo.php	(date 1684478870306)
@@ -20,6 +20,7 @@
 use Magento\Sales\Api\Data\CreditmemoItemInterface;
 use Magento\Sales\Model\Order;
 use Magento\Sales\Model\Order\Creditmemo as OrderCreditmemo;
+use Psr\Log\LoggerInterface;
 
 class Creditmemo extends Template
 {
@@ -53,12 +54,14 @@
         Config $configProvider,
         RewardsRepositoryInterface $rewardsRepository,
         QuoteResource $quoteResource,
+        LoggerInterface $logger,
         array $data = []
     ) {
         $this->coreRegistry = $registry;
         $this->configProvider = $configProvider;
         $this->rewardsRepository = $rewardsRepository;
         $this->quoteResource = $quoteResource;
+        $this->logger = $logger;
         parent::__construct($context, $data);
     }
 
@@ -84,7 +87,7 @@
     public function canRefundRewardPoints(): bool
     {
         $order = $this->getCreditmemo()->getOrder();
-
+        $this->logger->info('Go to sales/creditmemo with customer is guest: ' . $order->getCustomerIsGuest());
         if ($order->getCustomerIsGuest()) {
             return false;
         }
@@ -92,6 +95,11 @@
         return true;
     }
 
+    public function writeLog()
+    {
+        $this->logger->info('Go to creditmemo.phtml');
+    }
+
     /**
      * @return string
      */
Index: vendor/magento/module-reward/Block/Adminhtml/Sales/Order/Creditmemo/Reward.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-reward/Block/Adminhtml/Sales/Order/Creditmemo/Reward.php b/vendor/magento/module-reward/Block/Adminhtml/Sales/Order/Creditmemo/Reward.php
--- a/vendor/magento/module-reward/Block/Adminhtml/Sales/Order/Creditmemo/Reward.php	(revision 52e6cc80378f3e1121ce1d21e755ece04871568b)
+++ b/vendor/magento/module-reward/Block/Adminhtml/Sales/Order/Creditmemo/Reward.php	(date 1684479054910)
@@ -11,6 +11,8 @@
  */
 namespace Magento\Reward\Block\Adminhtml\Sales\Order\Creditmemo;
 
+use Psr\Log\LoggerInterface;
+
 /**
  * @api
  * @since 100.0.2
@@ -39,10 +41,12 @@
         \Magento\Backend\Block\Template\Context $context,
         \Magento\Framework\Registry $registry,
         \Magento\Reward\Helper\Data $rewardData,
+        LoggerInterface $logger,
         array $data = []
     ) {
         $this->_coreRegistry = $registry;
         $this->rewardData = $rewardData;
+        $this->logger = $logger;
         parent::__construct($context, $data);
     }
 
@@ -63,6 +67,7 @@
      */
     public function canRefundRewardPoints()
     {
+        $this->logger->info('Go to salesorder creditmemo' . $order->getCustomerIsGuest() . ' and rewards amount: ' . $this->getCreditmemo()->getOrder()->getRewardCurrencyAmount());
         if ($this->getCreditmemo()->getOrder()->getCustomerIsGuest()) {
             return false;
         }
Index: vendor/amasty/rewards/view/adminhtml/templates/sales/creditmemo.phtml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/amasty/rewards/view/adminhtml/templates/sales/creditmemo.phtml b/vendor/amasty/rewards/view/adminhtml/templates/sales/creditmemo.phtml
--- a/vendor/amasty/rewards/view/adminhtml/templates/sales/creditmemo.phtml	(revision 52e6cc80378f3e1121ce1d21e755ece04871568b)
+++ b/vendor/amasty/rewards/view/adminhtml/templates/sales/creditmemo.phtml	(date 1684478937930)
@@ -12,6 +12,7 @@
  * @var \Amasty\Rewards\Block\Adminhtml\Sales\Creditmemo $block
  */
 $isReadOnly = $block->isReadOnlyFields();
+$block->writeLog();
 ?>
 <?php if ($block->canRefundRewardPoints()): ?>
     <div class="admin__field admin__field-option field-refund-reward-points">
