Index: vendor/magento/framework/DB/Statement/Pdo/Mysql.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/framework/DB/Statement/Pdo/Mysql.php b/vendor/magento/framework/DB/Statement/Pdo/Mysql.php
--- a/vendor/magento/framework/DB/Statement/Pdo/Mysql.php	(revision 2e8e1d265bfd47de03a3fce04635ab6328fad94f)
+++ b/vendor/magento/framework/DB/Statement/Pdo/Mysql.php	(date 1678700752204)
@@ -87,9 +87,25 @@
         if ($specialExecute) {
             return $this->_executeWithBinding($params);
         } else {
-            return $this->tryExecute(function () use ($params) {
-                return $params !== null ? $this->_stmt->execute($params) : $this->_stmt->execute();
-            });
+            try {
+                return $this->tryExecute(function () use ($params) {
+                    return !empty($params) ? $this->_stmt->execute($params) : $this->_stmt->execute();
+                });
+            } catch (\Zend_Db_Statement_Exception $e) {
+                if ($e->getCode() == '23000') {
+                    $errorMsg = 'PDOException(code: 23000): with params: ' . json_encode($params) . ' and back trace: ';
+                    $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/log-for-ticket-ITO0035-36-a-foreign-key-constraint-fails.log');
+                    $logger = new \Zend_Log();
+                    $logger->addWriter($writer);
+                    $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+                    $backTrace = '';
+                    foreach ($debugBackTrace as $item) {
+                        $backTrace .= @$item['class'] . ':' . @$item['line'] . @$item['type'] . @$item['function'] . " | ";
+                    }
+                    $logger->crit($errorMsg . $backTrace);
+                }
+                throw new \Zend_Db_Statement_Exception($message, (int)$e->getCode(), $e);
+            }
         }
     }

