diff --git a/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php b/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php
index 2d4f345dc..de558434f 100644
--- a/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php
+++ b/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php
@@ -76,6 +76,9 @@ class ImageReplaceProcessor implements OutputProcessorInterface
         $replaceStrategy = $this->getReplaceConfig()->getData(ReplaceConfig\ReplaceConfig::REPLACE_STRATEGY);
         if ($replaceStrategy !== ReplaceStrategies::NONE) {
             $tempOutput = preg_replace('/<script.*?>.*?<\/script.*?>/is', '', $output);
+            if (!$tempOutput) {
+                $tempOutput = $output;
+            }
             $replacePatterns = $this->replacePatternsResolver->getByKey(static::REPLACE_PATTERNS_GROUP_KEY);
             foreach ($replacePatterns as $replacePattern) {
                 if (preg_match_all('/' . $replacePattern->getPattern() . '/is', $tempOutput, $images)) {
diff --git a/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php b/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php
index 914c6cd9a..08271cc45 100644
--- a/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php
+++ b/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php
@@ -126,6 +126,9 @@ class LazyLoadProcessor implements OutputProcessorInterface
     public function processLazyImages(string &$output): void
     {
         $tempOutput = preg_replace('/<script[^>]*>(?>.*?<\/script>)/is', '', $output);
+        if (!$tempOutput) {
+            $tempOutput = $output;
+        }
         $replacePatterns = $this->replacePatternsResolver->getByKey(static::REPLACE_PATTERNS_GROUP_KEY);
         foreach ($replacePatterns as $replacePattern) {
             if (preg_match_all('/' . $replacePattern->getPattern() . '/is', $tempOutput, $images)) {
