Index: vendor/magento/framework/Stdlib/DateTime/Timezone.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/framework/Stdlib/DateTime/Timezone.php b/vendor/magento/framework/Stdlib/DateTime/Timezone.php
--- a/vendor/magento/framework/Stdlib/DateTime/Timezone.php	(revision d1112a92a47090836fbc9f634e9c1a0c9707f9b6)
+++ b/vendor/magento/framework/Stdlib/DateTime/Timezone.php	(date 1669348059458)
@@ -342,6 +342,15 @@
             }
         } else {
             if ($date->getTimezone()->getName() !== $this->getConfigTimezone()) {
+                $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+                $backTrace = '';
+                foreach ($debugBackTrace as $item) {
+                    $backTrace .= @$item['class'] . ':' . @$item['line'] . @$item['type'] . @$item['function'] . " | ";
+                }
+                $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/debug-ticket-ITO0016-36.log');
+                $logger = new \Zend_Log();
+                $logger->addWriter($writer);
+                $logger->crit('Trace error timezone : ' . $backTrace);
                 throw new LocalizedException(
                     new Phrase(
                         'The DateTime object timezone needs to be the same as the "%1" timezone in config.',
