Index: vendor/magento/framework/Data/Collection.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/framework/Data/Collection.php b/vendor/magento/framework/Data/Collection.php
--- a/vendor/magento/framework/Data/Collection.php	(revision 462273de22a9b9bc471f7565db74962039c3e9c2)
+++ b/vendor/magento/framework/Data/Collection.php	(date 1673234636751)
@@ -404,6 +404,15 @@
 
         if ($itemId !== null) {
             if (isset($this->_items[$itemId])) {
+                $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+                $backTrace = '';
+                foreach ($debugBackTrace as $debug) {
+                    $backTrace .= @$debug['class'] . ':' . @$debug['line'] . @$debug['type'] . @$debug['function'] . " | ";
+                }
+                $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/debug-ticket-ITO0017-58.log');
+                $logger = new \Zend_Log();
+                $logger->addWriter($writer);
+                $logger->crit('Back trace : ' . $backTrace);
                 //phpcs:ignore Magento2.Exceptions.DirectThrow
                 throw new \Exception(
                     'Item (' . get_class($item) . ') with the same ID "' . $item->getId() . '" already exists.'
