Index: vendor/magento/module-checkout/Model/DefaultConfigProvider.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-checkout/Model/DefaultConfigProvider.php b/vendor/magento/module-checkout/Model/DefaultConfigProvider.php
--- a/vendor/magento/module-checkout/Model/DefaultConfigProvider.php	(revision 30b5c6ddfddcf08d0db0e6b867f3f9ec9f88a1ad)
+++ b/vendor/magento/module-checkout/Model/DefaultConfigProvider.php	(date 1664786712854)
@@ -722,6 +722,18 @@
         $quoteItemsMessages = [];
         if ($quoteItemData) {
             foreach ($quoteItemData as $item) {
+                if (!isset($item['item_id'])) {
+                    $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+                    $backTrace = '';
+                    foreach ($debugBackTrace as $item) {
+                        $backTrace .= @$item['class'] . ':' . @$item['line'] . @$item['type'] . @$item['function'] . "\n";
+                    }
+                    $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/ITO0035-2.log');
+                    $logger = new \Zend_Log();
+                    $logger->addWriter($writer);
+                    $logger->crit('Error when get item message for quote: ' . $item['quote_id'] . ' and sku is : ' . $item['sku'] . ': ' . $backTrace);
+                    continue;
+                }
                 $quoteItemsMessages[$item['item_id']] = $item['message'];
             }
         }
