Index: vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php b/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php
--- a/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php	(revision 6d0a4266580dc58c8fbe78d14a5c7a4408afc1f3)
+++ b/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php	(date 1667888681872)
@@ -114,7 +114,10 @@

         if ($this->getReplaceConfig()->getData(ReplaceConfig\ReplaceConfig::IS_REPLACE_IMAGES)) {
             $tempOutput = preg_replace('/<script.*?>.*?<\/script.*?>/is', '', $output);
-
+            if (!$tempOutput) {
+                $tempOutput = $output;
+            }
+
             foreach ($this->regexPatterns as $patternKey => $regexConfig) {
                 $this->currentRegexPattern = $patternKey;
                 if (preg_match_all('/' . $this->getRegex($regexConfig) . '/is', $tempOutput, $images)) {
