diff --git a/vendor/magento/module-re-captcha-frontend-ui/view/frontend/web/js/reCaptcha.js b/vendor/magento/module-re-captcha-frontend-ui/view/frontend/web/js/reCaptcha.js
index ddd627f..c88dccd 100644
--- a/vendor/magento/module-re-captcha-frontend-ui/view/frontend/web/js/reCaptcha.js
+++ b/vendor/magento/module-re-captcha-frontend-ui/view/frontend/web/js/reCaptcha.js
@@ -59,6 +59,13 @@ define(
              * @returns {Boolean}
              */
             getIsInvisibleRecaptcha: function () {
+
+                if (this.settings ===
+
+                    void 0) {
+                    return false;
+                }
+
                 return this.settings.invisible;
             },
 
@@ -83,7 +90,9 @@ define(
                     widgetId,
                     parameters;
 
-                if (this.captchaInitialized) {
+                if (this.captchaInitialized || this.settings ===
+
+                    void 0) {
                     return;
                 }
 
