diff --git a/vendor/amasty/module-shop-by-base/Model/UrlBuilder/UrlModifier.php b/vendor/amasty/module-shop-by-base/Model/UrlBuilder/UrlModifier.php
index 3ee2rd..8349152 111644
--- a/vendor/amasty/module-shop-by-base/Model/UrlBuilder/UrlModifier.php
+++ b/vendor/amasty/module-shop-by-base/Model/UrlBuilder/UrlModifier.php
@@ -17,7 +17,7 @@
     /**
      * @var UrlModifierInterface[]
      */
-    private $urlModifiers;
+    private $urlModifiers = [];

     public function __construct(array $urlModifiers = [])
     {
