Index: vendor/magento/module-quote/Model/ShippingAddressManagement.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-quote/Model/ShippingAddressManagement.php b/vendor/magento/module-quote/Model/ShippingAddressManagement.php
--- a/vendor/magento/module-quote/Model/ShippingAddressManagement.php	(revision 8d98c6285fe7a76dbb9d0c0d9e143567de922e4c)
+++ b/vendor/magento/module-quote/Model/ShippingAddressManagement.php	(date 1653968234819)
@@ -99,6 +99,18 @@
         $quote->setShippingAddress($address);
         $address = $quote->getShippingAddress();
 
+        if (!$address->getFirstname() || !$address->getLastname() || !$address->getStreet()) {
+            $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            $backTrace = '';
+            foreach ($debugBackTrace as $item) {
+                $backTrace .= @$item['class'] . ':' . @$item['line'] . @$item['type'] . @$item['function'] . "\n";
+            }
+            $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/debug-ticket-548.log');
+            $logger = new \Zend_Log();
+            $logger->addWriter($writer);
+            $logger->crit('Missing data shipping adress when assign shipping address: ' . $backTrace);
+        }
+
         if ($customerAddressId === null) {
             $address->setCustomerAddressId(null);
         }
Index: vendor/magento/module-quote/Model/QuoteManagement.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-quote/Model/QuoteManagement.php b/vendor/magento/module-quote/Model/QuoteManagement.php
--- a/vendor/magento/module-quote/Model/QuoteManagement.php	(revision 8d98c6285fe7a76dbb9d0c0d9e143567de922e4c)
+++ b/vendor/magento/module-quote/Model/QuoteManagement.php	(date 1653968234815)
@@ -527,6 +527,19 @@
                     'email' => $quote->getCustomerEmail()
                 ]
             );
+
+            if (!$shippingAddress->getFirstname() || !$shippingAddress->getLastname() || !$shippingAddress->getStreet()) {
+                $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+                $backTrace = '';
+                foreach ($debugBackTrace as $item) {
+                    $backTrace .= @$item['class'] . ':' . @$item['line'] . @$item['type'] . @$item['function'] . "\n";
+                }
+                $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/debug-ticket-548.log');
+                $logger = new \Zend_Log();
+                $logger->addWriter($writer);
+                $logger->crit('Missing data shipping adress when submit quote: ' . $backTrace);
+            }
+
             $shippingAddress->setData('quote_address_id', $quote->getShippingAddress()->getId());
             $addresses[] = $shippingAddress;
             $order->setShippingAddress($shippingAddress);
Index: vendor/magento/module-quote/Model/QuoteRepository/SaveHandler.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-quote/Model/QuoteRepository/SaveHandler.php b/vendor/magento/module-quote/Model/QuoteRepository/SaveHandler.php
--- a/vendor/magento/module-quote/Model/QuoteRepository/SaveHandler.php	(revision 8d98c6285fe7a76dbb9d0c0d9e143567de922e4c)
+++ b/vendor/magento/module-quote/Model/QuoteRepository/SaveHandler.php	(date 1653968234823)
@@ -117,6 +117,19 @@
             $this->billingAddressPersister->save($quote, $billingAddress);
         }
 
+        $address = $quote->getShippingAddress();
+        if (!$address->getFirstname() || !$address->getLastname() || !$address->getStreet()) {
+            $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            $backTrace = '';
+            foreach ($debugBackTrace as $item) {
+                $backTrace .= @$item['class'] . ':' . @$item['line'] . @$item['type'] . @$item['function'] . "\n";
+            }
+            $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/debug-ticket-548.log');
+            $logger = new \Zend_Log();
+            $logger->addWriter($writer);
+            $logger->crit('Missing data shipping adress when save quote: ' . $backTrace);
+        }
+
         $this->processShippingAssignment($quote);
         $this->quoteResourceModel->save($quote->collectTotals());
 
Index: vendor/magento/framework/DB/Adapter/Pdo/Mysql.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php b/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php
--- a/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php	(revision 8d98c6285fe7a76dbb9d0c0d9e143567de922e4c)
+++ b/vendor/magento/framework/DB/Adapter/Pdo/Mysql.php	(date 1653968047446)
@@ -619,6 +619,24 @@
      */
     public function query($sql, $bind = [])
     {
+        if ((stripos($sql, "quote_address") !== false || stripos($sql, "sales_order_address") !== false) && in_array('shipping', $bind, true) &&
+            (stripos($sql, "insert") !== false || (stripos($sql, "update") !== false && in_array(NULL, $bind)))) {
+            $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/debug-ticket-548.log');
+            $logger = new \Zend_Log();
+            $logger->addWriter($writer);
+            $nl = "\n";
+            $message = 'SQL: ' . $sql . $nl;
+            if ($bind) {
+                $message .= 'BIND: ' . var_export($bind, true) . $nl;
+            }
+            $logger->info('Insert | Update query: ' . $message);
+            try {
+                throw new \Exception("trace");
+            } catch (\Exception $exception) {
+                $logger->info($exception->getTraceAsString());
+            }
+        }
+
         if (strpos(rtrim($sql, " \t\n\r\0;"), ';') !== false && count($this->_splitMultiQuery($sql)) > 1) {
             throw new \Magento\Framework\Exception\LocalizedException(
                 new Phrase("Multiple queries can't be executed. Run a single query and try again.")
Index: vendor/magento/module-checkout/Model/ShippingInformationManagement.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-checkout/Model/ShippingInformationManagement.php b/vendor/magento/module-checkout/Model/ShippingInformationManagement.php
--- a/vendor/magento/module-checkout/Model/ShippingInformationManagement.php	(revision 8d98c6285fe7a76dbb9d0c0d9e143567de922e4c)
+++ b/vendor/magento/module-checkout/Model/ShippingInformationManagement.php	(date 1653968234819)
@@ -170,6 +170,18 @@
             $address->setCustomerAddressId(null);
         }
 
+        if (!$address->getFirstname() || !$address->getLastname() || !$address->getStreet()) {
+            $debugBackTrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            $backTrace = '';
+            foreach ($debugBackTrace as $item) {
+                $backTrace .= @$item['class'] . ':' . @$item['line'] . @$item['type'] . @$item['function'] . "\n";
+            }
+            $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/debug-ticket-548.log');
+            $logger = new \Zend_Log();
+            $logger->addWriter($writer);
+            $this->logger->critical('Missing data shipping adress when save shipping information: ' . $backTrace);
+        }
+
         try {
             $billingAddress = $addressInformation->getBillingAddress();
             if ($billingAddress) {
