Index: vendor/amasty/rgrid/Model/DuplicateRuleProcessor.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/amasty/rgrid/Model/DuplicateRuleProcessor.php b/vendor/amasty/rgrid/Model/DuplicateRuleProcessor.php
--- a/vendor/amasty/rgrid/Model/DuplicateRuleProcessor.php	(revision 206ffe733938ec3d88d1b7e48f389b7686ca7d83)
+++ b/vendor/amasty/rgrid/Model/DuplicateRuleProcessor.php	(date 1652087284148)
@@ -33,12 +33,9 @@
         if (is_array($attributes)) {
             $attributes[self::COUNT_USAGE_COLUMN] = 0;
         } else {
-            $attributes->setCount(0);
+            $attributes->setData(self::COUNT_USAGE_COLUMN, 0);
         }
-        $rule->setExtensionAttributes($attributes);
 
-        $this->ruleRepository->save($rule);
-
-        return $rule;
+        return $this->ruleRepository->save($rule);
     }
 }
