diff -u a/vendor/magento/module-checkout/Model/Session.php b/vendor/magento/module-checkout/Model/Session.php
index 3ee2rd..8349152 100755
--- a/vendor/magento/module-checkout/Model/Session.php	2022-07-24 14:23:05.940813050 +0700
+++ b/vendor/magento/module-checkout/Model/Session.php	2022-07-24 14:01:27.415000000 +0700
@@ -385,9 +385,16 @@
             if ($this->getQuoteId()) {
                 $quote = $this->getQuote();
                 $quote->setCustomerIsGuest(0);
-                $this->quoteRepository->save(
-                    $customerQuote->merge($quote)->collectTotals()
-                );
+                try {
+                    $this->quoteRepository->save(
+                        $customerQuote->merge($quote)->collectTotals()
+                    );
+                } catch (\Exception $e) {
+                    $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/261-merge-quote.log');
+                    $logger = new \Zend_Log();
+                    $logger->addWriter($writer);
+                    $logger->crit('Error merge quote ' . $quote->getId() . ' '. $e->getMessage());
+                }
                 $newQuote = $this->quoteRepository->get($customerQuote->getId());
                 $this->quoteRepository->save(
                     $newQuote->collectTotals()
