Index: vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php b/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php
--- a/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php	(revision edb69cae2a962b68f5e4ba893e37b909a1e0ca1d)
+++ b/vendor/amasty/module-lazy-load/Model/Output/LazyLoadProcessor.php	(date 1655187260541)
@@ -93,6 +93,9 @@
     public function processLazyImages(&$output)
     {
         $tempOutput = preg_replace('/<script[^>]*>(?>.*?<\/script>)/is', '', $output);
+        if (!$tempOutput) {
+            $tempOutput = $output;
+        }
 
         if (preg_match_all('/' . self::IMAGE_REGEXP . '/is', $tempOutput, $images)) {
             $skipCounter = 1;
diff --git a/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php b/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php
--- a/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php	(revision 0e06497ceda5e2080441a63ad15a13de3ed92856)
+++ b/vendor/amasty/module-image-optimizer/Model/Output/ImageReplaceProcessor.php	(date 1658460575820)
@@ -74,6 +74,9 @@

         if ($this->getReplaceConfig()->getData(ReplaceConfig\ReplaceConfig::IS_REPLACE_IMAGES)) {
             $tempOutput = preg_replace('/<script.*?>.*?<\/script.*?>/is', '', $output);
+            if (!$tempOutput) {
+                $tempOutput = $output;
+            }

             if (preg_match_all('/' . static::IMAGE_REGEXP . '/is', $tempOutput, $images)) {
                 foreach ($images[0] as $key => $image) {
